isis/src/base/objs/UserInterface/UserInterface.cpp

```diff
@@ -405,6 +405,20 @@ namespace Isis {
   }
 
 
+/**
+ * This is used to display the updated parameters when an app is called with -last
+ * 
+ * @return QString returnString- the updated command line 
+ */
+QString UserInterface::buildNewCommandLine(){
+  QString returnString = "";
+  for(auto i:p_cmdline){
+    returnString += i;
+    returnString += " ";
+  }
+  return returnString;
+}
```

Comment:
testing this locally, it only seems to be dittoing the command you entered instead of using the previous command. 

I get: 

```shell
(isisdev) $ cam2map FROM=test.cub TO=test_lev2 PIXRES=mpp RESOLUTION=160
(isisdev) $ cam2map -last RESOLUTION=20
cam2map -last RESOLUTION=20 
```

Im pretty sure the expected output is: 


```shell
(isisdev) $ cam2map FROM=test.cub TO=test_lev2 PIXRES=mpp RESOLUTION=160
(isisdev) $ cam2map cam2map -last RESOLUTION=20
cam2map FROM=test.cub TO=test_lev2 PIXRES=mpp RESOLUTION=20 
```

Additionally, I think there might be a way to make a test for this? Would building two UsterInterfaces back to back, one with `-LAST`, trigger the expected code in a way that is GTest-able?

Reply:
Im pretty sure this change needs to happen in `LoadHistory` (https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/objs/UserInterface/UserInterface.cpp#L550 called from https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/objs/UserInterface/UserInterface.cpp#L667)

Reply:
gtestable: I don't think the gtests should ever work with or modify a developers personal files, accessing history once to save a command line then again to retrieve it would do just that. The only way I can see to test this would be to have a temp user, much like a temp dir/temp file, then the solution seems possible.

I don't believe LoadHistory is the correct place. It only loads the history. The current command line with the history needs to be spiced together before the cout happens.

Reply:
One way to get the command line after the two have been merged might be to use the child class IsisAml.CommandLine(pvl) to get the current command as a pvl then loop through each keyword and construct the command line.

Reply:
@scsides There seems to be some logic just for tests: https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/objs/UserInterface/UserInterface.cpp#L670 I wonder if we can just change that boolean to be true when in a GTest or something?